source/Application/Component/UserComponent.php

```diff
@@ -372,13 +370,7 @@ protected function resetPermissions()
      */
     public function changeUser()
     {
-        $blUserRegistered = $this->_changeUser_noRedirect();
-
-        if ($blUserRegistered === true) {
-            return 'payment';
-        } else {
-            return $blUserRegistered;
-        }
+        return $this->_changeUser_noRedirect() ? 'payment' : false;
```

Comment:
Would be great to have strict comparison in most cases (===). Could you change this?